### DISCUSSION TITLE
JSON post request handling

### DISCUSSION THREAD

**Discussion by Jetinho** [mathieucarbou/ESPAsyncWebServer#160] [Q&A]
Hi there, 

I am facing a problem trying to implement a post route for which I need to parse a JSON body (form submission).

When using the [handler to get ready to use JSON](https://github.com/mathieucarbou/ESPAsyncWebServer?tab=readme-ov-file#json-body-handling-with-arduinojson), it seems that my server does not receive properly the request.

Actually, on the browser I get a blank page, and I get no serial log at all.

I guess I must be doing something wrong but I can't manage to find out where.

Here I commented all the code within the handler, to be sure it does not interfere with the issue I'm facing.
```
  AsyncCallbackJsonWebHandler *handler = new AsyncCallbackJsonWebHandler("/rest/endpoint", [](AsyncWebServerRequest *request, JsonVariant &json)
                                                                         {
    Serial.println("ESP32 Web Server: New POST request received:");
    Serial.println("POST /setlist");
    JsonObject jsonObj = json.as<JsonObject>();

  //   String jsonString;
  //   serializeJson(jsonObj, jsonString);
  //   Serial.println("Received JSON:");
  //   Serial.println(jsonString);

    // Send response
    request->send(200, "application/json", "{\"status\":\"success\"}"); });
  server.addHandler(handler);
```

On the other hand,I do manage to handle the post request with the following code : 

```
  server.on("/setlist", HTTP_POST, [](AsyncWebServerRequest *request)
            {
      Serial.println("ESP32 Web Server: New POST request received:");
      Serial.println("POST /setlist");

      // Send response
      request->send(200, "application/json", "{\"status\":\"success\"}"); });
```

**Comment by mathieucarbou** [comment]
Is it possible that your request is not correctly formed ?
The handler expect your request to have the content-type set to application/json.

Can you try it with curl -v and print the output here ?

**Reply by mathieucarbou** [reply]
I have updated the sample to show you how it works : https://github.com/mathieucarbou/ESPAsyncWebServer/blob/main/examples/SimpleServer/SimpleServer.ino#L575-L585
